File: Lista de exercicio/projetoPYTHON/questao14/questao14.py
class Configuracao:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Configuracao, cls).__new__(cls)
        return cls._instance

def executar_questao14():
    config1 = Configuracao()
    config2 = Configuracao()
    
    print(config1 is config2) 

File: Lista de exercicio/projetoJAVA/src/projetojava/questao14.java
package projetojava;

class Configuracao {
    private static Configuracao instancia;

    private Configuracao() {
    }

    public static Configuracao getInstancia() {
        if (instancia == null) {
            instancia = new Configuracao();
        }
        return instancia;
    }
}


public class questao14 {
    public static void main(String[] args) {
        Configuracao config1 = Configuracao.getInstancia();
        Configuracao config2 = Configuracao.getInstancia();

        System.out.println(config1 == config2);
    }
}
